### ISSUE TITLE
SQLfluff completely breaks query when a SQL comment contains jinja + trailing quotation marks

### ISSUE THREAD

**Issue by jlorince** [sqlfluff/sqlfluff#6261]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Formatting a sql query that contains a sql comment that itself contains jinja totally breaks the query if there are quotation marks in the SQL comment adjacent to the jinja

### Expected Behaviour

Ideally, I would expect sqlfluff to ignore jinja altogether if it's in a SQL comment, but at the very least this should break the query

### Observed Behaviour

Some examples (all using bigquery as the dialect; haven't tested with other dialects):
Formatting the following (jinja wrapped in quotes):
```
-- a commen with '{{ some_jinja }}'
SELECT *, FROM table;
```
yields:
```
{{ some_jinja }}'
SELECT *, FROM table;
```
Also happens if there is only a quote _after_ the jinja, e.g.
```
-- a comment with {{ some_jinja }}'
SELECT *, FROM table;
```
yields
```
{{ some_jinja }}'
SELECT *, FROM table;
```
However, if there is only a quote _before_ the jinja, there are no issues, e.g.
```
-- a comment with '{{ some_more_jinja }}
SELECT * FROM table;
```
yields
```
-- a comment with '{{ some_more_jinja }}
SELECT *, FROM table;
```

### How to reproduce

Just run any of the examples through sqlfluff with the default config, using single quotes, double quotes, or backticks.

### Dialect

bigquery

### Version

same behavior on 3.0.2 and 3.2.0

### Configuration

Defaults; no custom config

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by kaidaniel** [comment]
On version `3.2.0`, I tried to reproduce, but the file was unchanged:

```console
$ cat << EOF > example.sql
{% set x = 'x' %}
-- comment '{{ x }}'
SELECT a FROM t;
EOF

$ sqlfluff fix example.sql -d bigquery
...
All finished!

$ cat example.sql
{% set x = 'x' %}
-- comment '{{ x }}'
SELECT a FROM t;
```

**Comment by keraion** [comment]
@jlorince Is there any additional information you have when trying to run `sqlfluff`? I was also unable to reproduce this output with the given sql/configuration.

**Comment by Ktl-XV** [comment]
I believe I found the same issue, it does not need to be a comment to trigger the bug.

From what I found the starting quote of a jinja variable gets removed when:
* The indenting is not correct (LT02)
* The jinja value is a directory attribute (fails both using dot and square brackets notation)

To reproduce in 3.2.2:

```sh
$ cat << EOF > .sqlfluff
[sqlfluff:templater:jinja:context]
foo = foo
bar = bar
EOF

$ cat << EOF > test.sql
SELECT
  '{{ foo.bar }}'
FROM baz
EOF

$ sqlfluff fix test.sql -d postgres

$ cat test.sql
```

Output:
```
==== finding fixable violations ====
== [test.sql] FAIL
L:   2 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
== [test.sql] FIXED
1 fixable linting violations found
SELECT
    {{ foo.bar }}'
FROM baz
```
